### PR TITLE
api endpoints added.

### PR DESCRIPTION
**Pull request by DilsherB**
## PR Highlights:

1. Create an API endpoint to list all posts for a user.
2. Create an API endpoint to list all comments for a user's post.
3. Create an API endpoint to add a comment to a post. Remember that the owner of the comment is the user that makes it; it should not be set by passing a parameter.
4. Your API endpoints should receive JSON and respond JSON as well.

### CODE REVIEW

**Review by aamirkhan2478 — COMMENTED** [review]
Hi @DilsherB, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request :red_circle: :red_circle: :red_circle: 

You have submitted a project, that hasn't any commit from your coding partner kindly tell your partner about his/her contribution to your code for now your partner's commits are not in your PR :slightly_smiling_face: 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by shella12 — COMMENTED** [review]
Hi 

#  Status: Invalidated :red_circle: 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that is pair programming activity but your partner did not contribute significantly in the project. Please make sure that your partner has an almost equal contribution in the project (Not limited to README UPDATE)

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: config/routes.rb [hunk 1]
```diff
@@ -1,10 +1,8 @@
...
   root 'users#index'
-  # get "users", to: "users#index"
-  # get "users/:id", to: "users#show"
-
-  resources :users, only: [:index, :show] do
+  
+  resources :users, only: [:index, :show, :destroy] do
     resources :posts, only: [:index, :show, :new, :create, :destroy] do
       resources :comments, only: [:new, :create, :destroy]
       resources :likes, only: [:create, :destroy]
```
Comment (jerryowusu) [inline config/routes.rb:8]: No API endpoints here that returns all post in JSON format, hence it is difficult to get the api values for all comments and posts (nothing shows for the api when you run the rails routes) . Kindly check and add accordingly.

**Review by jerryowusu — CHANGES_REQUESTED** [review]
Hi @Team

Good job so far!
Your project is looking fantastic so far :star_struck: 

However, there are few changes you need to make in order to move on to the next level :sunglasses:

so close


## Highlights
Neat code :+1: 
Used gitflow :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: config/routes.rb (hunk 1 above)
Comment (DilsherB) [reply config/routes.rb:8]: Hi @jerryowusu, can you help me in this regard

**Review by Qoosim — COMMENTED** [review]
#  *Status: Invalid Request* :red_circle: 

Hi @team, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.



## Invalid Code Review Request

You have submitted a project, that the linters tests do not run on its PR. Kindly check it and fix.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by nahnahdev — APPROVED** [review]
## Project Status: APPROVED :green_circle: 
Hi @DilsherB  and @adarachel  , 

Your project is complete!  There is nothing else to say other than... It's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights :fire: 
- Linters well setup and passing :heavy_check_mark: 
- Professional README :heavy_check_mark: 
- All API endpoints receive JSON and respond in a JSON format as well :heavy_check_mark: 
- Has API endpoint that returns all posts in JSON format :heavy_check_mark: 
- Has API endpoint that returns all comments for a post :heavy_check_mark: 
- Has API endpoint that adds' a comment to a post :heavy_check_mark: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**